mqtt/mqtt-broker/src/settings/mod.rs

```diff
@@ -145,13 +145,35 @@ impl Default for RetainedMessagesConfig {
     }
 }
 
-// TODO: apply settings
 #[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct SessionPersistenceConfig {
     file_path: String,
     #[serde(with = "humantime_serde")]
     time_interval: Duration,
-    unsaved_message_count: u32,
+}
+
+// TODO: &self?
+impl SessionPersistenceConfig {
+    pub fn new(file_path: String, time_interval: Duration) -> Self {
+        Self {
+            file_path,
+            time_interval,
+        }
+    }
+
+    pub fn file_path(&self) -> String {
+        self.file_path.clone()
+    }
+
+    pub fn time_interval(&self) -> Duration {
```

Comment:
What does this time_interval signify?

Reply:
Interval at which the persisting happens. I can rename to make it more clear.

Reply:
So the broker will now persist the session on shutdown and periodically at an interval specified by this argument? 
And previously it was doing so every 5 mins?

Reply:
The broker was persisting on shutdown before, the only thing that changes here is the path and the interval specified by the arg. Previously it was every 5 mins, yes.

Reply:
Also, at some point in time we need to talk about our persistence strategy for edge hub. Right now I'm not sure that 5m interval is useful strategy, and I would rather disable it if possible. I would rather let the broker loose the state on crash/hard shutdown, then risk the inconsistent session state.

Reply:
I'm confused. Are you disagreeing with the 5m interval only? If we disable this then the broker still maintains state on shutdown right?

Reply:
Yes, I agree with Vadim. Making it configurable is okay for now, but we need to revisit it before we release.